### PR TITLE
ADBDEV-5274: Eliminate unnecessary variables in the DecodeDateTime() function

### PR DESCRIPTION
**Pull request by RekGRpth**
Eliminate unnecessary variables in the DecodeDateTime() function

The *tzp variable was not utilized anywhere in the DecodeDateTime() code, so
these assignments were redundant. Some callers of DecodeTimezone() depended on
the *tzp being non-NULL within that function.

Extend DecodeTimezone() to accept a NULL value as a second parameter. Use that
NULL value in DecodeDateTime() to indicate that it should not be modified.
Eliminate the need for t and *tzp in DecodeDateTime() and simplify the code.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68100

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1230978

**Comment by BenderArenadata** [comment]
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1230970

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1230979

File: src/interfaces/ecpg/pgtypeslib/dt_common.c [hunk 1]
```diff
@@ -1864,8 +1866,6 @@ DecodeDateTime(char **field, int *ftype, int nf,
 	int			haveTextMonth = FALSE;
 	int			is2digits = FALSE;
 	int			bc = FALSE;
-	int			t = 0;
-	int		   *tzp = &t;
```
Comment (bandetto) [inline src/interfaces/ecpg/pgtypeslib/dt_common.c:1868]: Why did these variables exist in the first place? It seems like they aren't used for anything even before the patch.

**Review by bandetto — APPROVED** [review]

**Comment by bandetto** [comment]
Can you change the description and the title to:
```
Eliminate unnecessary variables in the DecodeDateTime() function

The *tzp variable was not utilized anywhere in the DecodeDateTime() code, so
these assignments were redundant. Some callers of DecodeTimezone() depended on
the *tzp being non-NULL within that function.

Extend DecodeTimezone() to accept a NULL value as a second parameter. Use that
NULL value in DecodeDateTime() to indicate that it should not be modified.
Eliminate the need for t and *tzp in DecodeDateTime() and simplify the code.
```

**Comment by RekGRpth** [comment]
> Can you change the description and the title to:
> 
> ```
> Eliminate unnecessary variables in the DecodeDateTime() function
> 
> The *tzp variable was not utilized anywhere in the DecodeDateTime() code, so
> these assignments were redundant. Some callers of DecodeTimezone() depended on
> the *tzp being non-NULL within that function.
> 
> Extend DecodeTimezone() to accept a NULL value as a second parameter. Use that
> NULL value in DecodeDateTime() to indicate that it should not be modified.
> Eliminate the need for t and *tzp in DecodeDateTime() and simplify the code.
> ```

changed

**Pull request closed without merging**